migrator/README.md

```diff
@@ -0,0 +1,261 @@
+
+```File Migration``` - input a text file location with one E.164 number per line along with the BCP-47 country code that corresponds to the numbers in the text file. All numbers in the text file that match available recipes will be migrated and a copy of the text file will be created with the updated numbers. The file path to the newly created text file will be outputted to the console.
+
+```Migration Report``` - find all phone numbers in a given input that were converted with given migration recipes. The report will also show the phone numbers that were already seen as valid before migrations as well as identify any numbers that were migrated into invalid formats based on known valid phone number ranges for the specified country code.
+
+```Custom Recipe Migrations``` - specify custom recipe files to use for migrations instead of the internal maintained file. Custom recipe migrations will not check if converted numbers are valid based on country formats. 
+
+```Web Application``` - As an alternative to using the tool from the command line, [click here](https://phone-number-migrator.uc.r.appspot.com/) to access the web application version. 
+
+## Required Parameters
+
+```(-n, --number | -f, --file)``` - the phone number input for a given migration. The number argument must be a single E.164 phone number. The file argument must be a path to a text file containing one E.164 phone number per line. For file migrations, the original text file will not be overwritten and instead a new text file will be created containing the migrated version of each phone number or the original phone number when a migration did not take place. Note: all phone numbers entered for migration will be sanitized by removing any whitespace, hyphens or curved brackets. If the number after this process is still not in E.164 format, a migration will not be able to be performed on it.
+
+```-c, --countryCode``` - the BCP-47 country code that corresponds to the given phone number input. Only recipes from this country will be queried to find matching recipes for inputted numbers. (E.g. 44 for United Kingdom, 1 for US, 84 for Vietnam)
+
+## Optional Parameters
+
+```-r, --customRecipe``` - the path to a csv file containing custom migration recipes which follow the standard recipes file format. When using custom recipes, validity checks on migrated numbers will not be performed.
+
+```-e, --exportInvalidMigrations``` - the boolean command line flag specifying that text files created after the migration process for standard file migration job should contain the migrated version of a phone number when a migration has taken place on it, regardless of whether the migration resulted in an invalid phone number. By default, a strict approach is used and this is not the case. Note: this flag has no effect on a --number migration or a migration using a --customRecipe. 
```

Comment:
"By default"
Maybe you could add:
original numbers that couldn't be migrated or were considered invalid will not be replaced (or something similar to this). Just because this is the first time we used the word 'strict' and I want to make sure it is clear.